src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs

```diff
@@ -0,0 +1,231 @@
+//
+
+using System;
+using System.IO;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.Contracts;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts.ExecuteRequests;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.DataStorage;
+using Microsoft.SqlTools.ServiceLayer.SqlContext;
+using Microsoft.SqlTools.ServiceLayer.Workspace;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.Utility;
+using System.Diagnostics;
+using System.Xml.Serialization;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
+{
+    /// <summary>
+    /// Service for executing queries
+    /// </summary>
+    public class QuerySettingsHelper
+    {
+        //strings for various "SET <option> ON/OFF" statements
+        private static string s_On = "ON";
+        private static string s_Off = "OFF";
+        private static string s_Low = "LOW";
+        private static string s_Normal = "NORMAL";
+        private static string s_SetNoCount = "SET NOCOUNT {0}";
+        private static string s_SetConcatenationNull = "SET CONCAT_NULL_YIELDS_NULL {0}";
+        private static string s_SetNumericAbort = "SET NUMERIC_ROUNDABORT {0}";
+        private static string s_SetXACTAbort = "SET XACT_ABORT {0}";
+        private static string s_SetArithAbort = "SET ARITHABORT {0}";
+        private static string s_SetFmtOnly = "SET FMTONLY {0}";
+        private static string s_SetRowCount = "SET ROWCOUNT {0}";
+        private static string s_SetLockTimeout = "SET LOCK_TIMEOUT {0}";
+        private static string s_SetTextSize = "SET TEXTSIZE {0}";
+        private static string s_SetQueryGovernorCost = "SET QUERY_GOVERNOR_COST_LIMIT {0}";
+        private static string s_SetDeadlockPriority = "SET DEADLOCK_PRIORITY {0}";
+        private static string s_SetTranIsolationLevel = "SET TRANSACTION ISOLATION LEVEL {0}";
+        private static string s_SetAnsiNulls = "SET ANSI_NULLS {0}";
+        private static string s_SetAnsiNullDefault = "SET ANSI_NULL_DFLT_ON {0}";
+        private static string s_SetAnsiPadding = "SET ANSI_PADDING {0}";
+        private static string s_SetAnsiWarnings = "SET ANSI_WARNINGS {0}";
+        private static string s_SetCursorCloseOnCommit = "SET CURSOR_CLOSE_ON_COMMIT {0}";
+        private static string s_SetImplicitTransaction = "SET IMPLICIT_TRANSACTIONS {0}";
+        private static string s_SetQuotedIdentifier = "SET QUOTED_IDENTIFIER {0}";
+
+        private QueryExecutionSettings settings;
+
+        public QuerySettingsHelper(QueryExecutionSettings settings)
+        {
+            this.settings = settings;
+        }
+
+        [XmlIgnore]
+        public string SetNoCountString
```

Comment:
Why these attributes? Are you planning on having this be serialized at all anyways?

Reply:
this class is copy\paste from SSMS so I left it basically as implemented there.